WWDC/SessionsTableViewController+SupportingTypesAndExtensions.swift

```diff
@@ -82,59 +84,137 @@ extension Array where Element == SessionRow {
     }
 }
 
-struct FilterResults {
+final class FilterResults {
 
     static var empty: FilterResults {
         return FilterResults(storage: nil, query: nil)
     }
 
-    /// This becomes an OperationQueue for aborting, tada
-    private static let searchQueue = DispatchQueue(label: "Search", qos: .userInteractive)
+    private let query: NSPredicate?
 
-    var query: NSPredicate? {
-        didSet {
-            searchResults = nil
-        }
-    }
+    private let storage: Storage?
+
+    private(set) var latestSearchResults: Results<Session>?
 
-    let storage: Storage?
+    private var disposeBag = DisposeBag()
+    private let nowPlayingBag = DisposeBag()
 
-    private var searchResults: Results<Session>?
+    private var observerClosure: ((Results<Session>?) -> Void)?
+    private var observerToken: NotificationToken?
 
     init(storage: Storage?, query: NSPredicate?) {
         self.storage = storage
         self.query = query
-    }
 
-    func results(with closure: @escaping (Results<Session>?) -> Void) {
+        if let appDelegate = NSApplication.shared.delegate as? AppDelegate {
```

Comment:
Oh no, it's back 😂

Reply:
Haha, yeah! We still need to tackle “now playing” UI. At least now the session gets removed when you play a different video.